deno/payloads/v10/autoModeration.ts

```diff
@@ -0,0 +1,181 @@
+ */
+export enum AutoModerationRuleTriggerType {
+	/**
+	 * Check if content contains words from a user defined list of keywords (Maximum of 3 per guild)
+	 */
+	Keyword = 1,
+	/**
+	 * Check if content contains any harmful links (Maximum of 1 per guild)
+	 */
+	HarmfulLink,
+	/**
+	 * Check if content represents generic spam (Maximum of 1 per guild)
+	 */
+	Spam,
+	/**
+	 * Check if content contains words from internal pre-defined wordsets (Maximum of 1 per guild)
+	 */
+	KeywordPreset,
+}
+
+/**
+ * https://discord.com/developers/docs/resources/auto-moderation#auto-moderation-rule-object-trigger-metadata
+ */
+export interface AutoModerationRuleTriggerMetadata {
+	/**
+	 * Substrings which will be searched for in content
+	 *
+	 * A keyword can be a phrase which contains multiple words. Wildcard symbols can be used to customize how each keyword will be matched
+	 *
+	 * See https://discord.com/developers/docs/resources/auto-moderation#auto-moderation-rule-object-keyword-matching-strategies
+	 *
+	 * Associated trigger type: {@link AutoModerationRuleTriggerType.Keyword}
+	 */
+	keyword_filter: string[];
+	/**
+	 * The internally pre-defined wordsets which will be searched for in content
+	 *
+	 * Associated trigger type: {@link AutoModerationRuleTriggerType.KeywordPreset}
+	 */
+	presets: AutoModerationRuleKeywordPresetType[];
+}
+
+/**
+ * https://discord.com/developers/docs/resources/auto-moderation#auto-moderation-rule-object-keyword-preset-types
+ */
+export enum AutoModerationRuleKeywordPresetType {
+	/**
+	 * Words that may be considered forms of swearing or cursing
+	 */
+	Profanity = 1,
+	/**
+	 * Words that refer to sexually explicit behavior or activity
+	 */
+	SexualContent,
+	/**
+	 * Personal insults or words that may be considered hate speech
+	 */
+	Slurs,
+}
+
+/**
+ * https://discord.com/developers/docs/resources/auto-moderation#auto-moderation-rule-object-event-types
+ */
+export enum AutoModerationRuleEventType {
+	/**
+	 * When a member sends or edits a message in the guild
+	 */
+	MessageSend = 1,
+}
+
+/**
+ * https://discord.com/developers/docs/resources/auto-moderation#auto-moderation-action-object-auto-moderation-action-structure
+ */
+export interface APIAutoModerationAction {
+	/**
+	 * The action type
+	 */
+	type: AutoModerationActionType;
+	/**
+	 * Additional metadata needed during execution for this specific action type
+	 *
+	 * Will only be omitted if the action type is {@link AutoModerationActionType.BlockMessage}
+	 */
+	metadata?: AutoModerationActionMetadata;
+}
+
+/**
+ * https://discord.com/developers/docs/resources/auto-moderation#auto-moderation-action-object-action-types
+ */
+export enum AutoModerationActionType {
+	/**
+	 * Blocks the content of a message according to the rule
+	 */
+	BlockMessage = 1,
+	/**
+	 * Logs user content to a specified channel
+	 */
+	SendAlertMessage,
+	/**
+	 * Timeout user for specified duration, which requires the `MODERATE_MEMBERS` permission
```

Comment:
This isn't accurate. To set an action to timeout, your bot user needs the permission

Reply:
I changed it, what do you think now?